app/models/nic/bmc.rb

```diff
@@ -39,6 +37,24 @@ def ensure_physical
     def enc_attributes
       @enc_attributes ||= (super + %w(username password provider))
     end
+
+    private
+
+    def get_proxy
+      if subnet.present?
+        proxy = subnet.proxies.find { |subnet_proxy| subnet_proxy.has_feature?('BMC') }
+      end
+      proxy ||= SmartProxy.with_features("BMC").first
+      proxy
+    end
+
+    def validate_bmc_proxy
+      return true unless managed?
+      return true if host && !host_managed?
+      unless get_proxy
+        errors.add(:base, 'There is no proxy with feature ''BMC'' set up. Please register a smart proxy with this feature.')
```

Comment:
This string needs extracting, and avoid using `''` inside as gettext probably won't extract it properly.